src/main/java/hudson/maven/AbstractMavenProcessFactory.java

```diff
@@ -302,7 +302,11 @@ public String call() throws IOException {
                 if (originalJdk == null) { // so we only try this once
                     for (Throwable t = x; t != null; t = t.getCause()) {
                         if (t instanceof UnsupportedClassVersionError || t instanceof SocketException) {
-                            listener.error("[JENKINS-18403] JDK 5 not supported to run Maven; retrying with slave Java and setting compile/test properties to point to " + jdk.getHome());
+                            listener.error("[JENKINS-18403][JENKINS-28294] JDK '" + jdk.getName() +
+                                    "' not supported to run Maven projects; retrying with slave Java and setting compile/test properties to point to " +
+                                    jdk.getHome() +
+                                    ". Maven projects have to be launched with a Java version greater or equal to the one used by the slave. " +
```

Comment:
Pretty sure it's just a Maven version capable of running Jenkins. I.e. if your slave is running on Java 8, you can configure a JDK 7 here.